base/deprecated.jl

```diff
@@ -1689,14 +1689,14 @@ end
 # are deprecated in base/libgit2/types.jl.
 
 # deprecate ones/zeros methods accepting an array as first argument
-@deprecate ones(a::AbstractArray, ::Type{T}, dims::Tuple) where {T} fill!(similar(a, T, dims), 1)
-@deprecate ones(a::AbstractArray, ::Type{T}, dims...) where {T}     fill!(similar(a, T, dims...), 1)
-@deprecate ones(a::AbstractArray, ::Type{T}) where {T}              fill!(similar(a, T), 1)
-@deprecate ones(a::AbstractArray)                                   fill!(similar(a), 1)
-@deprecate zeros(a::AbstractArray, ::Type{T}, dims::Tuple) where {T}  fill!(similar(a, T, dims), 0)
-@deprecate zeros(a::AbstractArray, ::Type{T}, dims...) where {T}      fill!(similar(a, T, dims...), 0)
-@deprecate zeros(a::AbstractArray, ::Type{T}) where {T}               fill!(similar(a, T), 0)
-@deprecate zeros(a::AbstractArray)                                    fill!(similar(a), 0)
+@deprecate ones(a::AbstractArray, ::Type{T}, dims::Tuple) where {T} fill!(similar(a, T, dims), one(T))
+@deprecate ones(a::AbstractArray, ::Type{T}, dims...) where {T}     fill!(similar(a, T, dims...), one(T))
+@deprecate ones(a::AbstractArray, ::Type{T}) where {T}              fill!(similar(a, T), one(T))
+@deprecate ones(a::AbstractArray)                                   fill!(similar(a), one(eltype(a)))
+@deprecate zeros(a::AbstractArray, ::Type{T}, dims::Tuple) where {T}  fill!(similar(a, T, dims), zero(T))
+@deprecate zeros(a::AbstractArray, ::Type{T}, dims...) where {T}      fill!(similar(a, T, dims...), zero(T))
+@deprecate zeros(a::AbstractArray, ::Type{T}) where {T}               fill!(similar(a, T), zero(T))
+@deprecate zeros(a::AbstractArray)                                    fill!(similar(a), zero(eltype(a)))
```

Comment:
With benefit of hindsight, ideally each of these deprecations would throw a longer depwarn, containing similar information to that in the news entry. I conjecture most users will see only the depwarn, and the suggestions in the news entry that are simpler than the depwarn are likely to be much better received than the general replacement. Thoughts? :)